File: src/main/java/uk/co/eelpieconsulting/feedlistener/model/Channel.java
package uk.co.eelpieconsulting.feedlistener.model;

import org.bson.types.ObjectId;

import com.google.code.morphia.annotations.Entity;
import com.google.code.morphia.annotations.Id;

@Entity("channels")
public class Channel {

	@Id
	ObjectId objectId;

	private String id, name, username;
	
	public Channel() {
	}
	
	public Channel(String id, String name, String username) {
		this.id = id;
		this.name = name;
		this.username = username;
	}

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}
	
	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}

	@Override
	public String toString() {
		return "Channel [id=" + id + ", name=" + name + ", username=" + username + "]";
	}
	
}

File: src/main/kotlin/uk/co/eelpieconsulting/feedlistener/model/Channel.kt
package uk.co.eelpieconsulting.feedlistener.model

import com.fasterxml.jackson.annotation.JsonInclude
import dev.morphia.annotations.*
import dev.morphia.utils.IndexType
import org.bson.types.ObjectId

@Entity("channels")
@JsonInclude(JsonInclude.Include.NON_NULL)
@Indexes(
    Index(
        fields = [Field(value = "username", type = IndexType.DESC), Field(value = "name", type = IndexType.DESC)],
        options = IndexOptions(unique = true)
    )
)

data class Channel(@Id val objectId: ObjectId, val id: String, val name: String, val username: String)